C0103 fixes 4, datastore and db_store changes

davalb opened the pull request:
This PR is dependent on #1110.

Renelvon commented:
Yes, it's working fine now. :) Please squash the 2 commits together and I am merging this.

Happy new year!

davalb commented:
Happy new year to you, too! Cool, I just squashed the commits and rebased it. I am finally getting a hang of all the git magic :)
Will rebase c0103-fixes-5 as soon as I see you merging this branch.

Renelvon commented:
LGTM, merging.
